src/lib/render-context/index.ts

```diff
@@ -6,11 +6,15 @@ interface RenderContextState {
   setCameraTransform: (transform: Matrix) => void
 }
 
-export const useRenderContext = createStore<RenderContextState>((set, get) => ({
-  camera_transform: compose(scale(100, 100, 0, 0)),
-  setCameraTransform: (transform: Matrix) =>
-    set({ camera_transform: transform }),
-}))
+export const createRenderContextStore = () =>
+  createStore<RenderContextState>((set, get) => ({
+    camera_transform: compose(scale(100, 100, 0, 0)),
+    setCameraTransform: (transform: Matrix) =>
+      set({ camera_transform: transform }),
+  }))
 
-export const useCameraTransform = () =>
-  useRenderContext((s) => s.camera_transform)
+export const useCameraTransform = () => {
+  const useRenderContext = createRenderContextStore()
```

Comment:
Hmm this isnt quite right, you should use a ContextProvider and ony create the store once, its a bit tricky to explain but i think this is creating a store every time the hook is called.

React has a builtin called createContext that gives you a Provider and allows you to use useContext(context), so that is another solution (cant remember if thats how i did it in the linked code)


```
  const useRenderContext = useContext(renderContext)


// somewhere else...
const [store] = useState(() => createContextStore()) // useMemo with no dependencies also works
<renderContext.Provider value={store}>

</renderContext.Provider>
```